spec/lib/mime_negotiation_patch_spec.rb

```diff
@@ -0,0 +1,55 @@
+# -*- encoding : utf-8 -*-
+require 'spec_helper'
+
+describe 'MimeNegotiation#formats', :type => :request do
+
+  class AnonymousController < ApplicationController
+    def hello
+      render :text => "Hello world #{request.formats.first.to_s}!"
+    end
+
+    def all
+      render :text => self.formats.inspect
+    end
+
+    def get_file
+      render :file => "#{Rails.root}/README.md", :layout => false
+    end
+  end
+
+  before do
+    @routes.draw do
+      get 'file'  => 'anonymous#get_file'
+      get 'all'   => 'anonymous#all'
+      get 'hello' => 'anonymous#hello'
+    end
+  end
+
+  it 'returns HTML given a */* Accept header' do
+    get '/hello', {}, { 'HTTP_ACCEPT' => '*/*' }
+    expect(response.body).to eq 'Hello world */*!'
+  end
+
+  it 'returns HTML given a js or */* Accept header' do
+    get '/hello', {}, { 'HTTP_ACCEPT' => 'text/javascript, */*' }
+    expect(response.body).to eq 'Hello world text/html!'
+  end
+
+  it 'returns javascript given a js or */* Accept header on xhr' do
```

Comment:
The rails spec says it returns HTML but this spec says it returns javascript

https://github.com/rails/rails/blob/406e2f4679835fc83dade66e71412ca8ec382602/actionpack/test/controller/new_base/content_negotiation_test.rb#L27

Is this a Rails 3 vs 4 difference?

Reply:
I _think_ that's a typo in the spec description, most likely introduced when duplicating the test from the block directly above for `xhr`